testnet/stacks-node/src/event_dispatcher.rs

```diff
@@ -1615,4 +1698,272 @@ mod test {
+            drop(stream); // Close the stream
+        });
+
+        // Set a timeout shorter than the sleep duration to force a timeout
+        let connection_timeout = Duration::from_secs(2);
+
+        // Attempt to connect, expecting a timeout error
+        let result = send_request(
+            "127.0.0.1",
+            addr.port(),
+            b"{}",
+            &Url::parse("http://127.0.0.1/").unwrap(),
+            connection_timeout,
+        );
+
+        // Assert that the result is an error, specifically a timeout
+        assert!(
+            result.is_err(),
+            "Expected a timeout error, got: {:?}",
+            result
+        );
+
+        if let Err(err) = result {
+            assert_eq!(
+                err.kind(),
+                std::io::ErrorKind::WouldBlock,
+                "Expected TimedOut error, got: {:?}",
+                err
+            );
+        }
+    }
+
+    fn start_mock_server(response: &str, client_done_signal: Receiver<()>) -> String {
+        // Bind to an available port on localhost
+        let listener = TcpListener::bind("127.0.0.1:0").expect("Failed to bind server");
+        let addr = listener.local_addr().unwrap();
+
+        debug!("Mock server listening on {}", addr);
+
+        // Start the server in a new thread
+        let response = response.to_string();
+        thread::spawn(move || {
+            for stream in listener.incoming() {
+                debug!("Mock server accepted connection");
+                let mut stream = stream.expect("Failed to accept connection");
+
+                // Read the client's request (even if we don't do anything with it)
+                let mut buffer = [0; 512];
+                let _ = stream.read(&mut buffer);
+                debug!("Mock server received request");
+
+                // Simulate a basic HTTP response
+                stream
+                    .write_all(response.as_bytes())
+                    .expect("Failed to write response");
+                stream.flush().expect("Failed to flush stream");
+                debug!("Mock server sent response");
+
+                // Wait for the client to signal that it's done reading
+                client_done_signal
+                    .recv()
+                    .expect("Failed to receive client done signal");
+
+                debug!("Mock server closing connection");
+
+                // Explicitly drop the stream after signaling to ensure the client finishes
+                drop(stream);
+                break; // Close after the first request
+            }
+        });
+
+        // Return the address of the mock server
+        format!("{}:{}", addr.ip(), addr.port())
+    }
+
+    fn parse_http_response(response: &str) -> &str {
+        let parts: Vec<&str> = response.split("\r\n\r\n").collect();
+        if parts.len() == 2 {
+            parts[1] // The body is after the second \r\n\r\n
+        } else {
+            ""
+        }
```

Comment:
This is kinda innefficient, `collect()` forces it to scan the entire response and allocate a `Vec`, but we only need to find the end of the header. Also will fail if body contains `\r\n\r\n`. How about this:

```suggestion
        response.splitn(2, "\r\n\r\n")
            .nth(1)
            .unwrap_or("")
```

Reply:
This code is removed in Jude's changeset.